lib/find_non_repeated_letter.rb

```diff
@@ -1,14 +1,14 @@
-# This method takes a string, str, and returns the first non-repeated letter in that string.
-# More specifically, it looks for the first letter that appears by itself.
-#
-# + "ddcdd" has 'c' as its first non-repeated letter, and thus returns 'c'
-# + "aabccd" has both 'b' and 'd' as non-repeating letters, but would return 'b' because that
-#   one occurs first
-# + "aabbaa" only has repeated letters (two pairs of 'a', and a pair of 'b'), and thus returns
-#   nil (since there does not exist a non-repeated letter)
-
-
 def find_first_non_repeated_letter(str)
-  #write your method here
-  return true
-end
+    str.split('').each do |x|
+        str.count(x)
+        if str.count(x) == 1
+            return x
```

Comment:
If a unique character is found, this `return x` line would end the method. It works beautifully :)

Like if `str` was `"Hello"` (so the first non-repeated character is `"H"`), then your method wouldn't even bother to check if `"e"` or the other subsequent letters were unique. It would have stopped running as soon as it found that `"H"` is unique. That's a good thing!

But keeping that in mind, the next large section of code in your method is a bit unnecessary. For example, consider this re-write of your assignment:

```ruby
def find_first_non_repeated_letter(str)
    str.split('').each do |x|
        str.count(x)
        if str.count(x) == 1
            return x
        end
    end

    return nil
end
```

I replaced the entire section chunk of your code with `return nil`. But remember that that line of code _will never even be executed_ if the `return x` is triggered.

So while your code _does_ work well, I wanted to show this to you. Think about it for a bit, and see if you have any questions. If you don't, reply with something like "Got it!" and I'll approve this assignment.